server/testdriver/core/src/main/java/org/infinispan/server/test/core/ForkedInfinispanServerDriver.java

```diff
@@ -40,16 +42,26 @@
    private static final int SHUTDOWN_TIMEOUT_SECONDS = 15;
    private final List<ForkedServer> forkedServers = new ArrayList<>();
    private final List<Path> serverHomes;
+   private final String serverDataPath;
 
    protected ForkedInfinispanServerDriver(InfinispanServerTestConfiguration configuration) {
       super(configuration, InetAddress.getLoopbackAddress());
       String globalServerHome = configuration.properties().getProperty(TestSystemPropertyNames.INFINISPAN_TEST_SERVER_DIR);
       if (globalServerHome == null || globalServerHome.isEmpty()) {
          throw new IllegalArgumentException("You must specify a " + TestSystemPropertyNames.INFINISPAN_TEST_SERVER_DIR + " property.");
       }
+      serverDataPath = System.getProperty(Server.INFINISPAN_SERVER_ROOT_PATH);
+      if (serverDataPath != null && !serverDataPath.trim().isEmpty() && !serverDataPath.contains("%d")) {
+         throw new IllegalStateException("Server root path should have the index. Add the %d regex to the path. Example: /path/to/server_%d");
+      }
       this.serverHomes = new ArrayList<>();
       Path src = Paths.get(globalServerHome).normalize();
-      for (int i = 0; i < configuration.numServers(); i++) {
+      // ISPN-13109
+      copyServer(src, serverDataPath != null ? 1 : configuration.numServers());
+   }
+
+   private void copyServer(Path src, int maxServers) {
+      for (int i = 0; i < maxServers; i++) {
          Path dest = Paths.get(CommonsTestingUtil.tmpDirectory(), UUID.randomUUID().toString());
```

Comment:
I know it's not part of this PR's changes, but why do we use a UUID here instead of just `i`?

Reply:
+1 I was going to raise this too - the use of random uuid just makes this require another set of matching. This could either be just monotonically increasing with some prefix. Perhaps this could even be named by the test case/suite that required this instance.

Reply:
In this case, we will need to remove all files under `/tmp/server-x` for each test.
If a test fails during the test suite execution we will lose the logs as well.

Reply:
If we keep `server-i` instead of `UUID` the server folder must be removed before each test. If a test fails in the middle of the test suite, we will lose the logs.

Reply:
We can't use a test case name because the driver doesn't have the test case class.